### PR TITLE
Added more test for event Handlers

### PR DESCRIPTION
**Pull request by montera82**
Added more test for event Handlers

### CODE REVIEW

File: pkg/handler/user_email_changed.go [hunk 1]
```diff
@@ -23,23 +24,36 @@ func NewUserEmailChanged(s pkg.UserStore) *UserEmailChanged {
...
 // Handle is the user email changed message handler.
 func (u *UserEmailChanged) Handle(ctx context.Context, m *message.Message) error {
-	defer m.Ack(true)
-
 	user := new(pkg.User)
 	if err := json.Unmarshal(m.Body, user); err != nil {
 		log.Printf("failed to unmarshal message body: %v", err)
+		if err := m.Ack(false); err != nil {
+			log.Printf("failed to ack message: %v", err)
+			return err
```
Comment (rafaeljesus) [inline pkg/handler/user_email_changed.go]: we can remove this line, let's return the first error from `json.Unmarshal`

File: pkg/handler/user_email_changed.go [hunk 2]
```diff
@@ -23,23 +24,36 @@ func NewUserEmailChanged(s pkg.UserStore) *UserEmailChanged {
...
-			log.Print("user not found")
-		} else {
-			log.Printf("failed to save user to store: %v", err)
+	err := u.store.Save(user)
+
+	switch err {
+	case nil:
+		log.Print("user status successfully changed")
+		if err := m.Ack(false); err != nil {
+			return fmt.Errorf("failed to ack message: %v", err)
```
Comment (rafaeljesus) [inline pkg/handler/user_email_changed.go:42]: here is correct to return the ack error given the operation was successfully

File: pkg/handler/user_email_changed.go [hunk 3]
```diff
@@ -23,23 +24,36 @@ func NewUserEmailChanged(s pkg.UserStore) *UserEmailChanged {
...
+		}
+		return nil
+	case pkg.ErrNotFound:
+		log.Print("user not found")
+		if err := m.Ack(false); err != nil {
+			return fmt.Errorf("failed to ack message: %v", err)
```
Comment (rafaeljesus) [inline pkg/handler/user_email_changed.go]: let's return the actual error e.g `err` and not `Ack`

File: pkg/handler/user_email_changed_test.go [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+func TestUserEmailChanged(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		scenario string
+		function func(*testing.T, *mock.UserStore, *mock.Acknowledger)
+	}{
+		{
+			scenario: "When valid payload is supplied, then should successfully save user",
```
Comment (rafaeljesus) [inline pkg/handler/user_email_changed_test.go]: let's remove the `scenario:` and `:function`  fields name

Comment (rafaeljesus) [inline pkg/handler/user_email_changed_test.go]: I'd rather to have `when` lowered

File: pkg/handler/user_email_changed_test.go [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+func testShouldSuccessfullyChangeUserEmail(t *testing.T, store *mock.UserStore, acker *mock.Acknowledger) {
+	store.SaveFunc = func(user *pkg.User) error {
+		if user.Email != "foo@mail.com" {
+			t.Fatal("unexpected email")
+		}
+		if user.Username != "foo" {
+			t.Fatal("unexpected username")
+		}
+
```
Comment (rafaeljesus) [inline pkg/handler/user_email_changed_test.go]: we can remove this empty file

File: pkg/handler/user_email_changed_test.go [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+		if multiple {
+			t.Fatal("unexpected multiple")
+		}
+		return nil
+	}
+
+	body := []byte(`{
+		"email": "foo@mail.com",
+		"username": "foo",
+        "status": "active"
```
Comment (rafaeljesus) [inline pkg/handler/user_email_changed_test.go]: we can format this a bit

File: pkg/handler/user_email_changed_test.go [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+	handler := NewUserEmailChanged(store)
+	err := handler.Handle(context.Background(), msg)
+
+	if err != nil {
+		t.Fatalf("expected error to be nil, but got %v", err)
+	}
+	if !store.SaveInvoked {
+		t.Fatal("expected store.save() to be called")
+	}
+
```
Comment (rafaeljesus) [inline pkg/handler/user_email_changed_test.go]: we can remove this empty line

File: pkg/handler/user_email_changed_test.go [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+	body := []byte(`{
+		"email": "foo@mail.com",
+		"username": "foo",
+        "status": "active"
+	}`)
+
+	msg := message.New(acker, body)
+	handler := NewUserEmailChanged(store)
+	err := handler.Handle(context.Background(), msg)
+
```
Comment (rafaeljesus) [inline pkg/handler/user_email_changed_test.go]: remove empty line also here

File: pkg/handler/user_status_changed.go [hunk 9]
```diff
@@ -21,25 +22,38 @@ func NewUserStatusChanged(s pkg.UserStore) *UserStatusChanged {
...
+	case nil:
+		log.Print("user status successfully changed")
+		if err := m.Ack(false); err != nil {
+			return fmt.Errorf("failed to ack message: %v", err)
+		}
+		return nil
+	case pkg.ErrNotFound:
+		log.Print("user not found")
+		if err := m.Ack(false); err != nil {
+			return fmt.Errorf("failed to ack message: %v", err)
```
Comment (rafaeljesus) [inline pkg/handler/user_status_changed.go]: see previous comments about returning error

File: pkg/handler/user_status_changed_test.go [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/rafaeljesus/srv-consumer/pkg"
+	"github.com/rafaeljesus/srv-consumer/pkg/mock"
+	"github.com/rafaeljesus/srv-consumer/pkg/platform/message"
+)
+
+func TestUserStatusChanged(t *testing.T) {
```
Comment (rafaeljesus) [inline pkg/handler/user_status_changed_test.go:13]: same about the previous test file

Comment (montera82) [reply pkg/handler/user_status_changed_test.go:13]: all done nw.

**Pull request merged**